models/cryptopunks/ethereum/cryptopunks_ethereum_punk_offer_events.sql

```diff
@@ -0,0 +1,67 @@
+{{ config(
+        alias ='punk_offer_events',
+        partition_by = ['evt_block_time_week'],
+        materialized = 'incremental',
+        file_format = 'delta',
+        incremental_strategy = 'merge',
+        unique_key = ['evt_block_time_week', 'evt_tx_hash', 'evt_index'] 
+        )
+}}
+
+select event_type
+        , punk_id
+        , from
+        , to
+        , eth_amount
+        , eth_amount * p.price as usd_amount
+        , evt_block_time
+        , evt_block_time_week
+        , evt_block_number
+        , evt_index
+        , evt_tx_hash
+from 
+(   select  'Offered' as event_type
+            , punkIndex as punk_id
+            , b.from 
+            , case when toAddress = '0x0000000000000000000000000000000000000000' then cast(NULL as varchar(5))
+                else toAddress end as to
+            , minValue/1e18 as eth_amount
+            , a.evt_block_time 
+            , date_trunc('week',a.evt_block_time) as evt_block_time_week
+            , a.evt_block_number
+            , a.evt_index
+            , a.evt_tx_hash
+    from {{ source('cryptopunks_ethereum','CryptoPunksMarket_evt_PunkOffered') }} a
+    inner join {{ source('ethereum','transactions') }} b on a.evt_tx_hash = b.hash 
```

Comment:
please add incremental filter within join conditions on this source

Reply:
i wasn't clear here either. there should be two filters:
1 - your prior where clause, that can be reverted back to table alias `a` is filtered properly
2 - an additional join condition on table alias `b` for block time incremental filter -- we need to filter ethereum tx's base table as much as possible

Reply:
Got it! I think I understand and fixed in the latest commits just now but let me know if it's still off!

Reply:
pushed a commit with a simplified approach

Reply:
Thank you!